src/rgw/rgw_file.cc

```diff
@@ -956,7 +956,9 @@ namespace rgw {
   RGWFileHandle::~RGWFileHandle() {
     /* in the non-delete case, handle may still be in handle table */
     if (fh_hook.is_linked()) {
-      fs->fh_cache.remove(fh.fh_hk.object, this, FHCache::FLAG_LOCK);
+      /* in this case, we are being called from a context which holds
+       * the partition lock */
+      fs->fh_cache.remove(fh.fh_hk.object, this, FHCache::FLAG_NONE);
```

Comment:
may be we should keep this as it is, and instead change RGWFileHandle::reclaim:
https://github.com/ceph/ceph/pull/20034/files#diff-b03934e90d5d464122f42ce5e12cca65R1014